src/context.ts

```diff
@@ -285,6 +285,24 @@ const setDefaultContentType = (contentType: string, headers?: HeaderRecord): Hea
   }
 }
 
+/**
+ * This function ensures that the string is safely encoded,
+ * and not double-encoded.
+ * The input string may or may not escaped by encodeURI.
+ */
+const safeEncodeURI = (str: string) => {
+  try {
+    // decodeURI will throw if str contains %
+    // In that case, we can assume that str is not encoded
+    return encodeURI(decodeURI(str))
```

Comment:
I think this implementation is good. Using the exception makes the code shorter.